sympy/stats/crv.py

```diff
@@ -218,7 +218,10 @@ def _sample_numpy(cls, dist, size, seed):
         """Sample from NumPy."""
 
         import numpy
-        rand_state = numpy.random.default_rng(seed=seed)
+        if seed is None or isinstance(seed, int):
+            rand_state = numpy.random.default_rng(seed=seed)
+        else:
+            rand_state = seed
```

Comment:
These 4 lines are duplicated several times. After refactoring hopefully they can end up in a superclass or a function or something.